src/ui/draggable/horizontally-draggable.ts

```diff
@@ -0,0 +1,21 @@
+import { Draggable, DraggableConstraint } from './draggable';
```

Comment:
I think in terms of library consistency, looking at the way I wrote the other components rather than extending Draggable, we would just make an config in Draggable like this.

I personally think this is much easier to follow for other developers.

```
new Draggable({
   "horizontal": true
})
```

I think for the most part, for these simpler components, I prefer keeping it simple than being too quick to subclass which can lead to prematurely complicating it.

Reply:
Done.